src/lazy/text/buffer.rs

```diff
@@ -2002,4 +2030,50 @@ mod tests {
             mismatch_sexp(input);
         }
     }
+
+    #[test]
+    fn test_match_blob() {
+        fn match_blob(input: &str) {
+            MatchTest::new(input).expect_match(match_length(TextBufferView::match_blob));
+        }
+        fn mismatch_blob(input: &str) {
+            MatchTest::new(input).expect_mismatch(match_length(TextBufferView::match_blob));
+        }
+        // Base64 encodings of utf-8 strings
+        let good_inputs = &[
+            // hello
+            "{{aGVsbG8=}}",
+            "{{  aGVsbG8=}}",
+            "{{aGVsbG8=  }}",
+            "{{\taGVsbG8=\n\n}}",
+            // hello!
+            "{{aGVsbG8h}}",
+            "{{  aGVsbG8h}}",
+            "{{aGVsbG8h  }}",
+            "{{  aGVsbG8h  }}",
+            // razzle dazzle root beer
+            "{{cmF6emxlIGRhenpsZSByb290IGJlZXI=}}",
```

Comment:
Add a test case with extra white space in the middle.
```suggestion
            "{{cmF6emxlIGRhenpsZSByb290IGJlZXI=}}",
            "{{ cm F6emx lIGRhenps  ZSByb29  0IGJlZ XI= }}",
```